### PR TITLE
Update global website architecture

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
cc @brson 

This PR divides pages into smaller one and create dropdown menus as shown below:

screenshot from 2016-09-23 02-09-03

https://guillaumegomez.github.io/rust-www/en-US/

### CODE REVIEW

**Comment by AndrewBrinker** [comment]
A couple of questions:

- Do the dropdowns work effectively on mobile devices?
- Why the changes to indentation. It makes following the diffs more difficult.
- Do you have / can you put up a site showing what this would look like live? It's difficult to judge the differences right now.

**Comment by GuillaumeGomez** [comment]
> Do the dropdowns work effectively on mobile devices?

I kept the possibility to click on the dropdown url to have the same content. So it shouldn't change anything for mobile users.

> Why the changes to indentation. It makes following the diffs more difficult.

The current indentation was ugly and broken (mix of tabs and spaces).

> Do you have / can you put up a site showing what this would look like live? It's difficult to judge the differences right now.

I tried to but github doesn't it wants to make it happen (even when removing the CNAME file). For now, the only thing I can propose is unfortunately to test by yourself. :-/

File: _layouts/pt-BR/basic.html [hunk 1]
```diff
@@ -1,50 +1,76 @@
...
+					<h2><a href="/pt-BR/contribute.html">Contribua</a></h2>
+				</li>
+			</ul>
+		</header>
+
+		{{ content }}
+
+		<footer>
+			<p>Our site in other languages:
+				<a href="/en-US/">Inglesa</a>,
```
Comment (lifthrasiir) [inline _layouts/pt-BR/basic.html]: I guess this should read "English" (as it's the only clue for those speaks English and mistakenly went to pt-BR pages).

File: _layouts/ru-RU/basic.html [hunk 2]
```diff
@@ -1,50 +1,76 @@
...
+					<h2><a href="/ru-RU/contribute.html">Делать вклад</a></h2>
+				</li>
+			</ul>
+		</header>
+
+		{{ content }}
+
+		<footer>
+			<p>Our site in other languages:
+				<a href="/en-US/">английский</a>,
```
Comment (lifthrasiir) [inline _layouts/ru-RU/basic.html]: Same for this. Should read as "English".

Comment (GuillaumeGomez) [inline _layouts/ru-RU/basic.html]: I think you're right. Thanks!

**Comment by GuillaumeGomez** [comment]
Updated.

**Comment by steveklabnik** [comment]
Okay, so I pulled this down locally and gave it a shot.

I think it's more important to talk about if this is change we want to make, before reviewing the code, so I haven't looked at the implementation at all.

I like the way that this looks, and it works well. Splitting things up can make sense. But I also worry a lot about the impact this will have on all the translations, especially the WIP ones. Maybe that's their problem for not landing the bits yet :) We don't have a lot of other outstanding PRs.

/cc @rust-lang/core

**Comment by alexcrichton** [comment]
Could this be pushed to a temp location to browse around and poke at the changes as well?

**Comment by GuillaumeGomez** [comment]
@alexcrichton: I said it above. Even if I remove the CNAME file, jekyll don't generate it for some (unknown) reason. If someone has another idea, I'll be gladly follow it.

**Comment by lifthrasiir** [comment]
I've put a merge of my `ko-KR` branch (only available at `ko-KR/` URLs, do not follow this commit) with this PR at [my test subdomain](http://rust-www-ko-kr.xyzzy.mearie.org/en-US/). Seems to work.

**Comment by GuillaumeGomez** [comment]
Okay, I've removed CNAME so the website should be working: https://guillaumegomez.github.io/en-US/ (but it's not like I already said).

**Comment by alexcrichton** [comment]
Right now that's a 404 :(

**Comment by GuillaumeGomez** [comment]
@alexcrichton:

> but it's not, like I already said

**Comment by GuillaumeGomez** [comment]
Ok, I fixed the front page (other pages/urls __won't__ work!). You can have access here: https://guillaumegomez.github.io/rust-www/en-US/

**Comment by nikomatsakis** [comment]
I have to admit I am usually fairly skeptical of drop-down menus but these don't feel too bad. I think I'd prefer if the big page remained (so I can just click on Documentation and get a full listing), and each menu items led to an anchor in that page. Would this also help preserve ongoing documentation efforts?

**Comment by GuillaumeGomez** [comment]
I'm very split on how to handle files. I really don't like the fact that we have so big files. Finding information in it is just way too difficult. Too much information kill information. However, You can still have full content by clicking the dropdown menu url.

**Comment by GuillaumeGomez** [comment]
@nikomatsakis: I actually think that splitting these huge files into smaller ones would help a lot any documentation improvement.

Otherwise, any other opinion on this PR?

**Comment by GuillaumeGomez** [comment]
I updated the PR while taking into account new languages. It'd be nice to have other feedbacks before I have to update once again please.

**Comment by brson** [comment]
I apologize for not reviewing this sooner. I know that's frustrating. And I know
you are enthusiastic about this and I want to accomodate that. Frankly though I
have been apprehensive about the amount of effor this change would entail.

I'm not certain what the goals are with this patch, but I believe that there
are specific pieces of information that ideally would be easier to find and
that at least some of these menu items facilitiate that. Today one has to click
on the header they suspect might contain the type of information they are
looking for, then scan the entire page for section headers, whereas in this
design those section headers are generally elevated all the way to the front
page.

I haven't looked at the code yet, just played with the page, and here are my
notes on the design:

- menu items don't correspond consistently to sections in the pages
  - 'community' contains 'getting in touch ...'
    but that is not a section on the community page, rather
    it is three different sections
  - 'community' contains 'friends', which is not a section
    on the community page
- some of the individual pages are insubstantial
  - e.g. do we need an entire page to contain the content
    "For resources in languages other than English, see the locale-specific links in rust-learning."
  - e.g. 'nightly and beta docs'
- some headers have menus, some don't, with no clear distinction
- 'contribute' is the largest section of the site
  and it doesn't have a menu
- the 'project policies' menu item is not in the same
  order as it is on the docs page
- there are both large single pages containing all the content of the entire
  sections and individual pages duplicating that content
- the menus don't adapt for mobile devices. clicking on them
  just takes you to the main section page; the menus can't be used even though
  they appear

These mostly amount to confusing inconsistencies and make me think that adding
menus here necessarily cascades into a rethinking of considerably more,
particularly how the pages are organized into sections, and which sections are
important enough to warrant elevation into front-page menus.

So I think there are a number of problems to solve here still, and I cannot
dedicate time to it myself. Nor am I confident that submenus are desirable, and
if they are that they should be dropdown menus. It could help to understand more
clearly the problem being solved, whether there are alternatives.

I don't want to just say 'no', but I also can't spend my own time on
this, and I suspect that arriving at a satisfactory result here will require
extensive iteration. It may be fruitful to think about the issues I've raised
above and do a round of design with some other community members enthusiastic
about website design, like @AndrewBrinker, but I can't promise that it will lead
to a successful patch, and that you won't be disappointed with the process.

Again, I realize this is a frustrating response and that I'm a bottleneck on
improving the website.

My inclination is to close and continue to hold off on any major changes.

**Comment by AndrewBrinker** [comment]
There are a lot of changes here, and it's difficult to assess them all at the same time, particularly without discussion of what we want the website to be like. So far, updates to the site have been essentially as-needed. If we're going to overhaul the site, a more careful and principled approach would be better.

This is something I've actually been thinking about recently (what the process for redesigning the Rust site might look like), and here are my thoughts:

1. Start with a review of what content currently exists on the site. What are the different pages, how do they relate to each other?
2. Discuss what the current issues with the site are that a redesign should address, and determine what content should be prioritized.
3. Develop a content strategy for both structuring current content and for handling potential future additions to the site.
4. Determine the requirements for the development of the site, re: accessibility and browser support.
5. Develop and iterate on prototype designs, soliciting community feedback.
6. Once all involved feel confident about the changes to the site, transition to the new design.

This process would also need to involve the developers of the existing translations, and we would need a plan in place to support both the new site design and the previous design, until all existing translations have been updated to fit the new design.

I am happy to be involved in a process like this, if other people are interested in making this happen.

**Comment by GuillaumeGomez** [comment]
Isn't it a point of a PR? I opened it in hope to get people feedbacks and from there starts an iterative improvement process.

**Comment by AndrewBrinker** [comment]
I think the issue with the PR is that there are already a lot of changes made that need to be reviewed, and it's difficult to say if these changes are desirable because we doesn't have a clear enough sense right now of how we want the website to look and work.

I am suggesting a more organized effort to redesign the website, likely with a small team of main contributors pushing the work, and with a variety of community feedback solicitation to answer questions about how people are using the Rust website, what material is most important to them, and what they feel we should be prioritizing in our content strategy and information architecture.

This would both be easier to manage, and I think the end result would be better.

**Comment by aturon** [comment]
Thanks for the PR!

I agree with the general sense that the web site is in need of a more comprehensive design overhaul, which needs an organized effort with community feedback.

In particular, we're in the process of trying to contract with a design agency, who would work with some community liaisons and follow something much like the process @AndrewBrinker described. We'd want to do this all with an eye toward long-term sustainability as well. We should have firmer news on this front in a couple of weeks.

In the meantime, as @brson is saying, we're avoiding making significant changes to the site design.

**Comment by AndrewBrinker** [comment]
@aturon: good to hear about the design agency. I remember hearing that was a possible plan, but it's good to know that it's moving forward. I'll definitely keep an eye out for news on that front.

**Comment by steveklabnik** [comment]
I'm going to give this a close. Thanks again @GuillaumeGomez .

**Pull request closed without merging**